### PR TITLE
Event Publisher(Kafka), basic Slack integration

### PR DESCRIPTION
**Pull request by matoszz**
This PR sets up an initial Kafka implementation using IBM's Sarama package, the associated Dockerfiles / compose files, as well as a load balanced configuration with Envoy. This is a decently incomplete, but functional, setup just to get things rolling. Additionally within this PR:
- adds a new `pkg/events` with a kafka starter and event publisher / consumer adapter to be implemented at a later date
- adds basic consumer with slack integration 
- lays down basic configuration for a load balanced setup via Envoy but only includes 1 broker and Zookeeper today

### CODE REVIEW

**Review by golanglemonade — COMMENTED** [review]
This looks good! I pushed a commit with some feedback, and have a few comments,  otherwise LTGM

File: cmd/cli/cmd/events/event_create.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+	var s []byte
+
+	eventType := viper.GetString("event.create.type")
+	if eventType == "" {
+		return datum.NewRequiredFieldMissingError("type")
+	}
+
+	userid := viper.GetStringSlice("event.create.userid")
+	eventjson := viper.GetString("event.create.metadata")
```
Comment (golanglemonade) [inline cmd/cli/cmd/events/event_create.go:56]:
```suggestion
	eventjson := viper.GetString("event.create.eventjson")
```

File: pkg/events/kafka/shared/envoy/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,101 @@
+ARG ENVOY_IMAGE="${ENVOY_IMAGE:-envoyproxy/envoy}"
```
Comment (golanglemonade) [inline pkg/events/kafka/shared/envoy/Dockerfile:1]: This is a duplicate of what is in `docker/shared/envoy/Dockerfile` - can we remove this one?

Comment (matoszz) [reply pkg/events/kafka/shared/envoy/Dockerfile:1]: yep - had just missed cleaning it up !

**Review by golanglemonade — APPROVED** [review]

**Review by golanglemonade — APPROVED** [review]

**Pull request merged**